### ISSUE TITLE
Excessive Event Listeners

### ISSUE THREAD

**Issue by jjrabbit** [tovic/color-picker#48]
Hey tovic,

I came across some behavior that may be potentially undesirable. I noticed that every time a user clicks on the color input then a new event listener is automatically created, which is good, unless the user keeps clicking on the same input. Closing the color picker only removes the last event listener.

Here is a demonstration in Chrome:

https://i.imgur.com/E8QdStm.gif

I made an attempt to try and fix this behavior by preventing it in the first place.

Here is some code I added to your create() function inside color-picker.js, and a new function to query the color picker.

``` javascript
// create
function create(first, bucket) {
  // ...
  // ...
  if (first) {
    .style.left = self.style.top = '-9999px';
      function click(e) {
      var t = e.target,
      is_source = t === source || closest(t, source) === source;
        if (is_source) {
          if (colorPickerExists())        // <--- ADDED THIS
            return;
          else
            create(), trigger("enter");
        } else {
          $.exit();
        }
      }
  // ...
  // ...
}

// new method
function colorPickerExists() {
  const colorPicker = document.querySelector('.color-picker');
  return colorPicker;
}
``` 

I think I'm supposed to do a pull request but I'm still figuring out how github works 😂

**Comment by tovic** [comment]
Good catch! Actually, there is a function to check the existence of the color picker pane [here](https://github.com/tovic/color-picker/blob/master/color-picker.js#L380-L382). But instead of searching the DOM node by its class (that is not maintainable), I simply check the existence of its parent node. If it does exists, it means the DOM node already appended to the document.

**Comment by tovic** [comment]
Or, what will happen if you do `create(1)` or create(true)` there?

The first argument is used to tell that it is the first creation, so, don&rsquo;t repeat&hellip;

**Comment by jjrabbit** [comment]
Ah, I saw the visible() method but apparently I was using it wrong earlier. Yea same premise but you're right checking for parent works much better. Anyway if I do:

````javascript
if (is_source) {
  if (visible())
    return;
  else
    create(), trigger("enter");
} else {
  // ..
````

it also works fine.

I'm not 100% sure if this is what you meant with the create(1), but if I do:

````javascript
if (is_source) {
  create(1), trigger("enter");
} else {
  // ..
````

then the color picker never launches.

The odd thing is I just tested your code in a fiddle and I'm unable to create excessive event listeners there.

Is anyone able to confirm this happens to them also or is it something wrong with my machine?

I tested this with color-picker.html and color-picker.js (1.4.1) off the main site.
